python/ray/tune/tests/test_cluster.py

```diff
@@ -356,25 +351,47 @@ def test_migration_checkpoint_removal(start_connected_emptyhead_cluster,
         },
         "checkpoint_freq": 2,
         "max_failures": 2,
-        "remote_checkpoint_dir": "s3://tune-test-cluster/",
+        "remote_checkpoint_dir": MOCK_REMOTE_DIR,
+        "sync_to_driver_fn": trainable_id == "__fake_remote",
     }
 
-    # Test recovery of trial that has been checkpointed
-    t1 = Trial(trainable_id, **kwargs)
-    runner.add_trial(t1)
-    runner.step()  # start
-    runner.step()  # 1 result
-    runner.step()  # 2 result and checkpoint
-    assert t1.has_checkpoint()
-    cluster.add_node(num_cpus=1)
-    cluster.remove_node(node)
-    cluster.wait_for_nodes()
-    shutil.rmtree(os.path.dirname(t1.checkpoint.value))
-
-    runner.step()  # Recovery step
-    for i in range(3):
-        if t1.status != Trial.TERMINATED:
-            runner.step()
+    # The following patches only affect __fake_remote.
```

Comment:
If there's all of this patching, instead of optimizing for the parametrize, can you just have a second test for `__fake_durable`?

Reply:
either way i think it doesnt really matter rn, my guess is that we will merge DurableTrainable functionality into Trainable or make other changes so i'll revisit these tests soon anyways

Reply:
It's harder to reason about when reading..